cmd/main.go

```diff
@@ -1,34 +1,20 @@
 package main
 
 import (
-	"fmt"
-	"os"
-	"os/signal"
-	"syscall"
-
 	"github.com/kyma-project/cli/cmd/kyma"
 	"github.com/kyma-project/cli/internal/cli"
+	"os"
 
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 )
 
 func main() {
-	setupCloseHandler()
-	command := kyma.NewCmd(cli.NewOptions())
+	fin := cli.NewFinalizer()
+	fin.SetupCloseHandler()
+	command := kyma.NewCmd(cli.NewOptions(fin))
```

Comment:
Since there is only one initialisation for the finalizer and no dynamic fields at this point in time, maybe we can move this code to `cli.NewOptions()`.

Reply:
Finalization of the program is an important step in the life cycle of the application, exactly like running the main cmd line loop. I think that hiding it in the `cli.NewOptions()` method is not the best way to do ;) But of course, I can do this

Reply:
Ok yes, having it explicitly can be a good reason to have it in the `main`.

Still it bothers me a little that it is a constructor parameter that might not be necessary many times (calling a constructor with `nil` params usually is a smell)

Reply:
Ok. I will move this to the NewOptions method ;)